packages/runtime/container-runtime/src/test/gc/gcTelemetryTracker.spec.ts

```diff
@@ -0,0 +1,737 @@
+			}
+		};
+
+		describe("Inactive events", () => {
+			inactiveOrSweepEventTests(
+				inactiveTimeoutMs,
+				"inactive",
+				"GarbageCollector:InactiveObject_Revived",
+				"GarbageCollector:InactiveObject_Changed",
+				"GarbageCollector:InactiveObject_Loaded",
+			);
+		});
+
+		describe("SweepReady events", () => {
+			beforeEach(() => {
+				injectedSettings[disableSweepLogKey] = true;
+			});
+
+			inactiveOrSweepEventTests(
+				sweepTimeoutMs,
+				"sweep",
+				"GarbageCollector:SweepReadyObject_Revived",
+				"GarbageCollector:SweepReadyObject_Changed",
+				"GarbageCollector:SweepReadyObject_Loaded",
+				false, // expectDeleteLogs
+			);
+		});
+
+		if (isSummarizerClient) {
+			describe("SweepReady events - with delete log", () => {
+				inactiveOrSweepEventTests(
+					sweepTimeoutMs,
+					"sweep",
+					"GarbageCollector:SweepReadyObject_Revived",
+					"GarbageCollector:SweepReadyObject_Changed",
+					"GarbageCollector:SweepReadyObject_Loaded",
+					true, // expectDeleteLogs
+				);
+			});
+		}
+	};
+
+	describe("Summarizer client", () => {
+		clientTypeTests(true /* isSummarizerClient */);
+	});
+
+	describe("Interactive client", () => {
+		clientTypeTests(false /* isSummarizerClient */);
+	});
+
+	describe("gcUnknownOutboundReferences telemetry", () => {
+		const unknownReferenceEventName = "GarbageCollector:gcUnknownOutboundReferences";
+		const currentGCData: IGarbageCollectionData = { gcNodes: {} };
+		let previousGCData: IGarbageCollectionData;
+		let explicitReferences: Map<string, string[]>;
+
+		beforeEach(() => {
+			telemetryTracker = createTelemetryTracker(
+				true /* enableSweep */,
+				true /* isSummarizerClient */,
+			);
+
+			currentGCData.gcNodes["/"] = [nodes[0]];
+			currentGCData.gcNodes[nodes[0]] = [nodes[1]];
+			currentGCData.gcNodes[nodes[1]] = [nodes[0], nodes[2]];
+			currentGCData.gcNodes[nodes[2]] = [nodes[1], nodes[3]];
+			currentGCData.gcNodes[nodes[3]] = [nodes[0]];
+
+			previousGCData = cloneGCData(currentGCData);
+			explicitReferences = new Map();
+		});
+
+		it("does not log gcUnknownOutboundReferences when there are no new references", async () => {
+			telemetryTracker.logIfMissingExplicitReferences(
+				currentGCData,
+				previousGCData,
+				explicitReferences,
+				mc.logger,
+			);
+
+			mockLogger.assertMatchNone(
+				[
+					{
+						eventName: unknownReferenceEventName,
+					},
+				],
+				"There should be no gcUnknownOutboundReferences event",
+			);
+		});
+
+		it("logs gcUnknownOutboundReferences when there are unknown data store references", async () => {
+			const gcNodeId = nodes[0];
+			const gcRoutes = [nodes[2], nodes[3]];
+			currentGCData.gcNodes[gcNodeId] = gcRoutes;
+
+			telemetryTracker.logIfMissingExplicitReferences(
+				currentGCData,
+				previousGCData,
+				explicitReferences,
+				mc.logger,
+			);
+
+			mockLogger.assertMatch(
+				[
+					{
+						eventName: unknownReferenceEventName,
+						gcNodeId,
+						gcRoutes: JSON.stringify(gcRoutes),
```

Comment:
We're logging the routes?  I think that needs to be tagged as codeArtifact since it could include an aliased datastore.

Reply:
Yeah, you are right. I will do this in a follow up.

Reply:
https://dev.azure.com/fluidframework/internal/_workitems/edit/401/